plugins/docs-tools/skills/commit-analyst/scripts/gather_changes.py

```
@@ -0,0 +1,650 @@
+
+# ---- Extraction: GitLab (REST API, unauthenticated) ----
+
+def _gitlab_api_get(host, endpoint):
+    """GET a GitLab REST API endpoint. Returns parsed JSON or None."""
+    url = f"https://{host}/api/v4{endpoint}"
+    token = os.environ.get("GITLAB_TOKEN") or os.environ.get("GITLAB_PRIVATE_TOKEN")
+    req = Request(url)
+    if token:
+        req.add_header("PRIVATE-TOKEN", token)
+    try:
+        with urlopen(req, timeout=30) as resp:
```

Ruff (0.15.11)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)